src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.cs

```diff
@@ -218,6 +264,8 @@ private static bool IsComment(string line)
             return false;
         }
 
+        private static bool IsGlobalFileName(string fileName) => Path.GetFileName(fileName).Equals(UserGlobalConfigName, StringComparison.OrdinalIgnoreCase);
```

Comment:
This method is called a number of times and every time is an allocation. There also doesn't appear to be a case where we call it zero times. Consider just calculating this once and storing it in a field.